build_tools/cmake/setup_ccache.sh

```diff
@@ -0,0 +1,30 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# This file must be *sourced* not executed.
+
+set -euo pipefail
+
+IREE_READ_REMOTE_CCACHE="${IREE_READ_REMOTE_CCACHE:-1}"
+IREE_WRITE_REMOTE_CCACHE="${IREE_WRITE_REMOTE_CCACHE:-0}"
+if (( ${IREE_WRITE_REMOTE_CCACHE} == 1 && ${IREE_READ_REMOTE_CCACHE} != 1 )); then
+  echo "Can't have 'IREE_WRITE_REMOTE_CCACHE' (${IREE_WRITE_REMOTE_CCACHE})" \
+       " set without 'IREE_READ_REMOTE_CCACHE' (${IREE_READ_REMOTE_CCACHE})"
+fi
+
+if (( IREE_READ_REMOTE_CCACHE == 1 )); then
+  export CCACHE_REMOTE_STORAGE="http://storage.googleapis.com/iree-sccache/ccache"
```

Comment:
`iree-sccache` intentional here, or do you want to use the name `iree-ccache` instead? (From the PR description it looks like you evaluated sccache but decided to go with ccache instead)

Reply:
Yeah but I'd already created the bucket before I realized that ccache now had remote support as well. sccache stands for "Shared Compilation Cache" so it still works. It didn't seem worth creating a new bucket, but I can if this is confusing

Reply:
Okay, sure. Keeping the name is fine.